pkg/fileutils/fileutils_linux.go

```diff
@@ -0,0 +1,18 @@
+package fileutils
+
+import (
+	"os"
+
+	"github.com/docker/docker/pkg/system"
+)
+
+// CurrentUserIsOwner checks whether the current user is the owner of the given
+// file.
+func CurrentUserIsOwner(f string) (bool, error) {
```

Comment:
I don't really like the function returning an `error` if it's always `nil`. The only potential use of the `error` value is to distinguish between a failure to `Stat` and unmatched `uid`s when the returned `bool` is false:
- If there's no use case for this, then the function should only return `bool`
- If there's a use case for this (spoiler: I don't think so), then one code path should return `false, err`